src/Rules/In.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Oshomo\CsvUtils\Rules;
+
+use Oshomo\CsvUtils\Contracts\ArrayParameterizedRuleInterface;
+use Oshomo\CsvUtils\Contracts\ValidationRuleInterface;
+
+class In implements ValidationRuleInterface, ArrayParameterizedRuleInterface
+{
+    /**
+     * Determines if the validation rule passes. This is where we do the
+     * actual validation. If the validation passes return true else false.
+     *
+     * @param mixed $value
+     */
+    public function passes($value, array $parameters, array $row): bool
+    {
+        if (null === $value || '' === $value) {
+            return true;
+        }
+
+        return in_array($value, $parameters);
+    }
+
+    /**
+     * Get the validation error message. Specify the message that should
+     * be returned if the validation fails. You can make use of the
+     * :attribute and :value placeholders in the message string.
+     */
+    public function message(): string
+    {
+        return 'The :attribute value :value does not exist in :allowed_values on line :line.';
+    }
+
+    /**
+     * Should return an array of the allowed parameters.
+     * The allowed parameters should be tokenized string
+     * e.g :min, :max, :first, :last etc.
+     */
+    public function allowedParameters(): array
```

Comment:
I am struggling to see why this is needed. The `$parameters` is already an array.